storage/perfschema/pfs_account.h

```diff
@@ -62,22 +63,22 @@ struct PFS_ALIGNED PFS_account : PFS_connection_slice
 public:
   inline void init_refcount(void)
   {
-    PFS_atomic::store_32(& m_refcount, 1);
+    m_refcount.store(1);
   }
 
   inline int get_refcount(void)
   {
-    return PFS_atomic::load_32(& m_refcount);
+    return m_refcount.load();
   }
 
   inline void inc_refcount(void)
   {
-    PFS_atomic::add_32(& m_refcount, 1);
+    m_refcount.fetch_add(1);
   }
 
   inline void dec_refcount(void)
   {
-    PFS_atomic::add_32(& m_refcount, -1);
+    m_refcount.fetch_sub(1);
   }
```

Comment:
Would it make sense to use [release-acquire ordering](https://en.cppreference.com/w/cpp/atomic/memory_order#Release-Acquire_ordering) at least in these operations, instead of continuing the use of the more expensive sequential consistent memory order? (For stores, according to https://www.cl.cam.ac.uk/~pes20/cpp/cpp0xmappings.html using a weaker ordering would seem to always avoid some overhead.)

Reply:
I think since it's a refactoring patch, we shouldn't demand ordering improvement in this PR/MDEV. However yes, pfs may have some improvements after some well-thinking. But before improving the performance schema performance, I'd fix the bugs.

Here's your favorite one: https://github.com/MariaDB/server/blob/d821fd7fab2723b84f774aaa4882acb86b35769d/storage/perfschema/pfs_setup_actor.cc#L88

Also don't think it's a good place to demand a bugfix

Reply:
@FooBarrior Can you file bugs for those suggestions and note the MDEV numbers here? I agree that we should not extend the scope of this pull request.